src/select.js

```diff
@@ -513,29 +513,39 @@
       return ctrl.placeholder;
     };
 
-    var containerSizeWatch;
     ctrl.sizeSearchInput = function(){
-      var input = _searchInput[0],
-          container = _searchInput.parent().parent()[0];
+      var input                    = _searchInput[0],
+          container                = _searchInput.parent().parent()[0],
+          matchedElementsContainer = angular.element(input).parent().children()[0];
+
       _searchInput.css('width','10px');
-      var calculate = function(){
-        var newWidth = container.clientWidth - input.offsetLeft - 10;
-        if(newWidth < 50) newWidth = container.clientWidth;
-        _searchInput.css('width',newWidth+'px');
+
+      var getSizeProperty = function(element, property){
+
+        return parseInt(window.getComputedStyle(element, null).getPropertyValue(property) , 10);
```

Comment:
This is a very expensive check - may be a good idea to find a way to cache the information of `window.getComputedStyle` and find a better way to hook into it.